message_sender/tests.py

```diff
@@ -1918,3 +1920,70 @@ def test_channel_default_post_save(self):
         channel.save()
 
         self.assertEqual(Channel.objects.filter(default=True).count(), 1)
+
+
+class TestManagementCommands(AuthenticatedAPITestCase):
+
+    def make_identity_lookup(self, msisdn='+27123', identity='test-identity'):
+        identity = {
+            "msisdn": msisdn,
+            "identity": identity,
+        }
+        lookup = IdentityLookup.objects.create(**identity)
+        return lookup
+
+    def test_update_identity_no_argument(self):
+        self.make_outbound()
+        self.make_inbound('1234', from_addr='+27123')
+        self.make_identity_lookup()
+        call_command('update_identity_field')
+
+        d = Outbound.objects.last()
+        self.assertEqual(str(d.to_addr), "")
+        self.assertEqual(str(d.to_identity), "test-identity")
+
+        i = Inbound.objects.last()
+        self.assertEqual(str(i.from_addr), "")
+        self.assertEqual(str(i.from_identity), "test-identity")
+
+    def test_update_identity_by_id(self):
+        self.make_outbound()
+        self.make_inbound('1234', from_addr='+27123')
+        self.make_identity_lookup()
+        call_command('update_identity_field', '--loop', 'ID')
+
+        d = Outbound.objects.last()
+        self.assertEqual(str(d.to_addr), "")
+        self.assertEqual(str(d.to_identity), "test-identity")
+
+        i = Inbound.objects.last()
+        self.assertEqual(str(i.from_addr), "")
+        self.assertEqual(str(i.from_identity), "test-identity")
+
+    def test_update_identity_by_msg(self):
+        self.make_outbound()
+        self.make_inbound('1234', from_addr='+27123')
+        self.make_identity_lookup()
+        call_command('update_identity_field', '--loop', 'MSG')
+
+        d = Outbound.objects.last()
+        self.assertEqual(str(d.to_addr), "")
+        self.assertEqual(str(d.to_identity), "test-identity")
+
+        i = Inbound.objects.last()
+        self.assertEqual(str(i.from_addr), "")
+        self.assertEqual(str(i.from_identity), "test-identity")
+
+    def test_update_identity_by_sql(self):
+        self.make_outbound()
+        self.make_inbound('1234', from_addr='+27123')
+        self.make_identity_lookup()
+        call_command('update_identity_field', '--loop', 'SQL')
+
+        d = Outbound.objects.last()
+        self.assertEqual(str(d.to_addr), "")
+        self.assertEqual(str(d.to_identity), "test-identity")
+
+        i = Inbound.objects.last()
+        self.assertEqual(str(i.from_addr), "")
+        self.assertEqual(str(i.from_identity), "test-identity")
```

Comment:
It's not strictly necessary, but a nice model I've seen for these kinds of tests. You write the test once in a base class, then for each of the specific types you create a class that inherits from the base class, and all that class does is change the specific command that's called.

That way, you only need to write each test once, instead of 3 times. We can do that, because we expect the output of each of these functions to be the same.